### PR TITLE
Adding camel language support for Camel XML DSL

### LINKED ISSUE #3528: Provide Language Server for Apache Camel
Is your feature related or already mentioned on [the wishlist](https://github.com/emacs-lsp/lsp-mode/discussions/2892)? No

There is a [Language Server for Apache Camel](https://github.com/camel-tooling/camel-language-server). It would be nice having it built-in. [Apache Camel](https://camel.apache.org/) is an Open Source integration framework that empowers you to quickly and easily integrate various systems consuming or producing data. It is one of the top active projects in Apache foundation.

The server is written in Java so I guess a good example on how to integrate it is the [Language Server for xml integration](https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-xml.el).
The Camel LSP jar is available here https://repo1.maven.org/maven2/com/github/camel-tooling/camel-lsp-server/1.5.0/camel-lsp-server-1.5.0.jar

A specificity of Apache Camel is that it is working with several languages: java, xml, yaml. As a first iteration, we can imagine to enable it for a single language.
There are several settings potentially available ([Camel Catalog version](https://github.com/camel-tooling/camel-language-server#specific-version-of-camel-catalog), [Camel runtime provider](https://github.com/camel-tooling/camel-language-server#specific-runtime-provider-of-camel-catalog), [Additional Camel component](https://github.com/camel-tooling/camel-language-server#additional-camel-components) but as a first iteration i think we can ignore all of them. None are mandatory.

Questions: Are there some tests to implement to incorporate the additional language? or the tests are only checking core features of lsp-mode?

### PR DESCRIPTION
**Pull request by Navyashree0923**
Fixes #3528 
Please note that this pull request is covering only Camel XML DSL. So, Java, Yaml DSL are not covered. The language server settings are not configurable. This plan to be implemented in another iteration.

### CODE REVIEW

File: clients/lsp-camel.el [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ :group 'lsp-camel
+ :type 'file
+ :package-version '(lsp-mode . "8.0.0"))
+
+(defcustom lsp-camel-jar-download-url
+ (format "https://repo1.maven.org/maven2/com/github/camel-tooling/camel-lsp-server/%s/%s" lsp-camel-jar-version lsp-camel-jar-name)
+ "Automatic download url for lsp-camel."
+ :type 'string
+ :group 'lsp-camel
+ :package-version '(lsp-mode . "8.0.0"))
```
Comment (yyoncho) [inline clients/lsp-camel.el:33]: can you change all of these to 8.0.1. Also, can you format the buffer via `C-x h M-x indent-region`?

Comment (Navyashree0923) [reply clients/lsp-camel.el]: Hello @yyoncho  
Done with the changes, can you please review it.

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Comment by Navyashree0923** [comment]
Thank you @yyoncho

**Pull request merged**